lib/src/migrators/module.dart

```diff
@@ -1227,12 +1248,33 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   /// longest matching prefix removed.
   ///
   /// Otherwise, returns [name] unaltered.
-  String _unprefix(String name) {
-    var isPrivate = name.startsWith('-');
-    var unprivateName = isPrivate ? name.substring(1) : name;
-    var prefix = _prefixFor(unprivateName);
-    if (prefix == null) return name;
-    return (isPrivate ? '-' : '') + unprivateName.substring(prefix.length);
+  ///
+  /// If [forcePublic] is true, this will ensure that the name is always a
+  /// public identifier.
+  String _unprefix(String name, {bool forcePublic = false}) {
+    bool isPrivate;
+    String withoutPrefix;
+
+    // Allow users to pass prefixes that either do or do not include leading
+    // dashes/underscores. As usual, the longest prefix wins, so we check for
+    // ones that do include them first.
+    var prefix = _prefixFor(name);
+    if (prefix != null) {
+      isPrivate = false;
+      withoutPrefix = name.substring(prefix.length);
+    } else {
+      isPrivate = _isPrivate(name);
+      var unprivateName = isPrivate ? _privateToPublic(name) : name;
+      prefix = _prefixFor(unprivateName);
+      if (prefix == null)
+        return isPrivate && forcePublic ? unprivateName : name;
+      withoutPrefix = unprivateName.substring(prefix.length);
+    }
+
+    if (_isPrivate(withoutPrefix)) {
+      withoutPrefix = _privateToPublic(withoutPrefix);
+    }
+    return (isPrivate && !forcePublic ? '-' : '') + withoutPrefix;
```

Comment:
I thought because of this line, here we return '-' but we should return the original private prefix.

But from testing, it seems there is some magic running after that already and it keeps the original private prefix somehow